aspnetcore/grpc/index.md

```diff
@@ -0,0 +1,185 @@
+
+[!code-cs[](~/tutorials/grpc/grpc-start/samples/GrpcStart/GrpcGreeter.Server/Services/GreeterService.cs?highlight=10,12-18)]
+
+For client side assets, a concrete client type is generated. The gRPC calls in the `.proto` file are translated to methods on the concrete type which can be called. For the `greet.proto` example described above, a concrete `GreeterClient` type is generated that contains a `SayHello` method that can be called to initiate a gRPC to the server.
+
+[!code-cs[](~/tutorials/grpc/grpc-start/samples/GrpcStart/GrpcGreeter.Client/Program.cs?highlight=19,21)]
+
+By default, both server and client assets are generated for each `.proto` file included in the `<ProtoBuf>` item group. To ensure only the server assets are generated in a server project, the `GrpcServices` attribute is set to `Server`.
+
+[!code-xml[](~/tutorials/grpc/grpc-start/samples/GrpcStart/GrpcGreeter.Server/GrpcGreeter.Server.csproj?highlight=8)]
+
+Similarly the attribute is set to `Client` in client projects:
+
+[!code-xml[](~/tutorials/grpc/grpc-start/samples/GrpcStart/GrpcGreeter.Client/GrpcGreeter.Client.csproj?highlight=10)]
+
+## Add gRPC services to your ASP.NET Core app
+
+### Prerequisite packages
+
+To obtain the gRPC APIs for ASP.NET Core projects, the [Grpc.AspNetCore.Server](https://www.nuget.org/packages/Grpc.AspNetCore.Server) package must be added to the project. The [Google.Protobuf](https://www.nuget.org/packages/Google.Protobuf/) must also be added to ensure the APIs for protobuf messages are available.
+
+[!code-xml[](~/tutorials/grpc/grpc-start/samples/GrpcStart/GrpcGreeter.Server/GrpcGreeter.Server.csproj?highlight=13-14)]
+
+### Configure `Startup`
+
+gRPC is enabled in the `Startup.ConfigureServices` method through the `AddGrpc` method:
+
+[!code-cs[](~/tutorials/grpc/grpc-start/samples/GrpcStart/GrpcGreeter.Server/Startup.cs?highlight=18)]
+
+Each gRPC service is added to the ASP.NET Core routing pipeline through the `MapGrpc` method in `Startup.Configure`:
+
+[!code-cs[](~/tutorials/grpc/grpc-start/samples/GrpcStart/GrpcGreeter.Server/Startup.cs?highlight=29-32)]
+
+Since ASP.NET Core middlewares and features share the routing pipeline, an application can be configured to serve additional request handlers such as MVC controllers in parallel with the configured gRPC services.
+
+## Integration with ASP.NET Core APIs
+
+gRPC services have full access to the ASP.NET Core features such as Dependency Injection (DI) and Logging. For example, the service implementation can resolve a logger from the DI container via the constructor:
+
+```csharp
+public class GreeterService : Greeter.GreeterBase
+{
+    public GreeterService(ILogger<GreeterService> logger)
```

Comment:
What is the recommended type to inject by users? `ILogger<T>` or `ILoggerFactory`?

Reply:
I see ILogger<T> being used in the docs more often: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-2.2#create-logs